### ISSUE TITLE
SUCO: `ModuleNotFoundError: No module named 'rosdep2', bash_return_code: 1`

### ISSUE THREAD

**Issue by 130s** [kinu-garage/hut_10sqft#1100]
```
:
Cmd: ['/usr/bin/pip', 'install', 'rosdep', '--break-system-packages']. See https://stackoverflow.com/questions/75602063 for the risk of passing '--break-system-packages' option.
subprocess: About to execute the cmd: /home/gm130s/.local/bin/rosdep init             
output: , error: Traceback (most recent call last):
  File "/home/gm130s/.local/bin/rosdep", line 5, in <module>              
    from rosdep2.main import rosdep_main            
ModuleNotFoundError: No module named 'rosdep2', bash_return_code: 1
subprocess: About to execute the cmd: /home/gm130s/.local/bin/rosdep update                              
output: , error: ERROR: no sources directory exists on the system meaning rosdep has not yet been initialized.
                                                    
Please initialize your rosdep with
        sudo rosdep init, bash_return_code: 1 
subprocess: About to execute the cmd: /usr/bin/apt update
```

**Comment by 130s** [comment]
On brya, I can't repro by calling `rosdep` outside of SUCO...
```
noodler@penguin:~$ which rosdep
/home/noodler/.local/bin/rosdep
noodler@penguin:~$ rosdep init
ERROR: cannot create /etc/ros/rosdep/sources.list.d:
        [Errno 13] Permission denied: '/etc/ros'
```

**Comment by 130s** [comment]
But then uninstalling attempt fails with the same issue as https://github.com/kinu-garage/hut_10sqft/issues/1099, which I thought I addressed by https://github.com/kinu-garage/hut_10sqft/pull/1098..

```
 pip uninstall rosdep
error: externally-managed-environment

× This environment is externally managed
╰─> To install Python packages system-wide, try apt install
    python3-xyz, where xyz is the package you are trying to
    install.
    
    If you wish to install a non-Debian-packaged Python package,
    create a virtual environment using python3 -m venv path/to/venv.
    Then use path/to/venv/bin/python and path/to/venv/bin/pip. Make
    sure you have python3-full installed.
    
    If you wish to install a non-Debian packaged Python application,
    it may be easiest to use pipx install xyz, which will manage a
    virtual environment for you. Make sure you have pipx installed.
    
    See /usr/share/doc/python3.11/README.venv for more information.

note: If you believe this is a mistake, please contact your Python installation or OS distribution provider. You can override this, at the risk of breaking your Python installation or OS, by passing --break-system-packages.
hint: See PEP 668 for the detailed specification.
```

**Comment by 130s** [comment]
I've spent a good amount of time to catch with the usecases and the thoughts around Python's virtual environment. https://hynek.me/articles/virtualenv-lives/ that seems to be written one of the long time PSF fellow was particularly insightful.

That said one of the usecases SUCO has to cover is that there are two different kinds of users for the packages SUCO installs, one is SUCO itself and the other is the user of the OS. The former is the common userbase that the dependency of an application considers (so in this context SUCO may be similar to a package manager that has its own dependency AND installs what the OS user uses).

For that usecase I'm not yet sure what's the best way to move foward.
All the potential issues those articles I've read are warning sound real. I just realized that I've been avoiding to rely on `pip` packages as much as possible, which may be a side effect of how ROS treats `pip` as a developer-only option. But now in SUCO I'm inclined to use `pip` for some functionalities, and once I start doing that "those issues" may start bothering my computer usage if untreated.

Thought `pipx` could be a solution. But [stackoverflow.com#78229687](https://stackoverflow.com/questions/78229687/what-is-the-difference-between-pipx-and-using-pip-install-inside-a-virtual-envir) mentions it blocks an `import` access to the pkgs installed by it.

----

Those are general thoughts. For the very issue in this ticket, I'm switching to add ros apt source and have `rosdep` installed from `apt`. As the need for ROS apt source provides more pkgs than rosdep, some of those may not be available from `pypi`, it just makes sense for the usecases of my computers.
